docs/standard/native-interop/tutorial-comwrappers.md

```diff
@@ -0,0 +1,430 @@
+
+```csharp
+var getter = (IDemoGetType)rcw;
+var store = (IDemoStoreType)rcw;
+
+string msg = "hello world!";
+store.StoreString(msg.Length, msg);
+Console.WriteLine($"Setting string through wrapper: {msg}");
+
+value = demo.GetString();
+Console.WriteLine($"Get string through managed object: {value}");
+
+msg = msg.ToUpper();
+demo.StoreString(msg.Length, msg.ToUpper());
+Console.WriteLine($"Setting string through managed object: {msg}");
+
+value = getter.GetString();
+Console.WriteLine($"Get string through wrapper: {value}");
+```
+
+Since your `ComWrapper` subclass was designed to support `CreateObjectFlags.UniqueInstance`, you can clean up the Native Object Wrapper immediately instead of waiting for a GC to occur.
+
+```csharp
+(rcw as IDisposable)?.Dispose();
+```
+
+## Additional considerations
+
+**Native AOT** &ndash; Ahead-of-time (AOT) compilation provides improved startup cost as JIT compilation is avoided. Removing the need for JIT compilation is also often required on some platforms. Supporting AOT was a goal of the `ComWrappers` API, but any wrapper implementation must be careful not to inadvertantly introduce cases where AOT breaks down, such as using reflection. The `Type.GUID` property is an example of where reflection is used, but in a non-obvious way. The `Type.GUID` property uses reflection to inspect the type's attributes and then potentially the type's name and containing assembly in order to generate its value.
+
+**Source generation** &ndash; Most of the code that's needed for COM interop and a `ComWrappers` implementation can likely be autogenerated by some tooling. Source for both types of wrappers could be generated given the proper COM definitions &ndash; for example, Type Library (TLB), IDL, or a Primary Interop Assembly (PIA).
+
+**Global registration** &ndash; Since the `ComWrappers` API was designed as a new phase of COM interop, it needed to have some way to partially integrate with the existing system. There are globally impacting static methods on the `ComWrappers` API that permit registration of a global instance for various support. These methods are designed for `ComWrappers` instances that are expecting to provide comprehensive COM interop support in all cases &ndash; akin to the built-in COM interop system.
+
+[**Reference Tracker support**][api_referencetracker] &ndash; This support is primary used for WinRT scenarios and represents an advanced scenario. For the majority of `ComWrapper` implementations, either a `CreateComInterfaceFlags.TrackerSupport` or `CreateObjectFlags.TrackerObject` flag should throw a <xref:System.NotSupportedException>. If you'd like to enable this support, perhaps on a Windows or even non-Windows platform, it is highly recommended to reference the [C#/WinRT tool chain][repo_cswinrt].
```

Comment:
>  or even non-Windows platform

Does this mean that COM Wrappers work on all platforms? I think that should be mentioned somewhere in the doc. Actually it should mention it either way - whether it's Windows specific or not.

Reply:
>Actually it should mention it either way - whether it's Windows specific or not.

That is part of the API definition. It is cross platform now. I believe we default to "it is cross-platform unless otherwise stated" &ndash; that seems to be the .NET promise. We updated the API metadata in .NET 6 for cross-platform.

Reply:
It is surprising to me (in a good way). Mainly because I always think about COM as Windows-only. This is definitely COM, so I expected it to be Windows only. That's why I think it's worth pointing it out.